### ISSUE TITLE
ora is not interactive in empty node application on serve command

### ISSUE THREAD

**Issue by wizardnet972** [nrwl/nx#17107]
### Current Behavior

I create a workspace with `ts` preset and empty node application with `esbuild`.

I installed `ora@5.4.1` (the non-esm version) to the project and change my `main.ts` I put this code:
```
import ora from 'ora';
console.log('Hello World');

const spinner = ora('Loading unicorns').start();

setTimeout(() => {
  spinner.color = 'yellow';
  spinner.text = 'Loading rainbows';
}, 1000);
```

And then start the project: `nx serve cli`.

The results spinner doesn't spin:

<img width="732" alt="image" src="https://github.com/nrwl/nx/assets/4997825/28dcc487-9569-47ac-821b-96c231124aa3">

### Expected Behavior

The spinner should work. Because in node application without nx it works. also when I build and start with node it also works. I guess that nx wraps the std output and can't allow it to be interactive. 

### GitHub Repo

https://github.com/wizardnet972/nx-node-ora

### Steps to Reproduce

1. clone the repo and run npm install
2. run the command: `nx serve cli`
3. you will see that not interactive.

### Nx Report

```shell
yarn run v1.22.19
$ /Users/wizardnet972/nx-sandbox/nx-node-ora/node_modules/.bin/nx report

 >  NX   Report complete - copy this into the issue template

   Node   : 18.13.0
   OS     : darwin arm64
   npm    : 9.3.0
   Hasher : Native

   nx                 : 16.2.1
   @nx/js             : 16.2.1
   @nx/jest           : 16.2.1
   @nx/linter         : 16.2.1
   @nx/workspace      : 16.2.1
   @nx/devkit         : 16.2.1
   @nx/esbuild        : 16.2.1
   @nx/eslint-plugin  : 16.2.1
   @nx/node           : 16.2.1
   @nrwl/tao          : 16.2.1
   typescript         : 5.0.4

✨  Done in 0.77s.
```
```


### Failure Logs

_No response_

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

**Comment by chrisfrancis27** [comment]
Hitting this exact same thing running an `@nx/js:node` lib. The stdout text defaults to bold red, it can be changed by `chalk.yellow(...)` etc. but the Ora spinner is just a static red hyphen. Invoking the node script directly rather than via Nx works as expected.

**Comment by AgentEnder** [comment]
This should be fixed by the native command runner in Nx 17.3, which has already been merged. I'm going to close this out as such.
